### PR TITLE
Remove unused locals and methods in System.Private.Xml

### PR DESCRIPTION
**Pull request by MattKotsenas**
Remove unused locals and private methods in System.Private.Xml.

This fixes part of #30457

### CODE REVIEW

**Comment by krwq** [comment]
Please do not merge this PR until we're done with nullable annotations for XML. Merging may cause unnecessary conflicts with work in progress which we're currently targeting for 5.0, you can track the effort here: https://github.com/dotnet/runtime/issues/2339

**Comment by stephentoub** [comment]
@MattKotsenas, can you rebase this on master, now that the PRs @krwq described have been merged?  Thanks.

**Comment by MattKotsenas** [comment]
@stephentoub, rebased. @krwq, can you take another look when you have a moment? Thanks!

File: src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializationReader.cs
Comment (stephentoub) [inline src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializationReader.cs:190]:
```suggestion
            _r.NameTable.Add(XmlSchema.Namespace);
            _r.NameTable.Add(UrtTypes.Namespace);
```

File: src/libraries/System.Private.Xml/src/System/Xml/Xsl/Xslt/CompilerScopeManager.cs
Comment (stephentoub) [inline src/libraries/System.Private.Xml/src/System/Xml/Xsl/Xslt/CompilerScopeManager.cs:278]:
```suggestion
                        GetName(ref _records[prev], out prevPrefix, out _);
```

File: src/libraries/System.Private.Xml/src/System/Xml/Xsl/Xslt/XsltLoader.cs
Comment (stephentoub) [inline src/libraries/System.Private.Xml/src/System/Xml/Xsl/Xslt/XsltLoader.cs:847]:
```suggestion
            ParseCollationAttribute(3);
```

File: src/libraries/System.Private.Xml/src/System/Xml/Xsl/Xslt/XsltLoader.cs
Comment (stephentoub) [inline src/libraries/System.Private.Xml/src/System/Xml/Xsl/Xslt/XsltLoader.cs:2140]:
```suggestion
            ParseCollationAttribute(3);
```

File: src/libraries/System.Private.Xml/tests/XmlDocument/XmlDocumentTests/CreateCDataSectionTests.cs
Comment (stephentoub) [inline src/libraries/System.Private.Xml/tests/XmlDocument/XmlDocumentTests/CreateCDataSectionTests.cs:40]:
```suggestion
            AssertExtensions.Throws<ArgumentException>(null, () => cdata.OuterXml);
```

File: src/libraries/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs
Comment (stephentoub) [inline src/libraries/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs:745]:
```C#
Assert.Throws<InvalidOperationException>(() => (TypeWithByteProperty)serializer.Deserialize(stream));
```

File: src/libraries/System.Private.Xml/tests/XmlNodeReader/System.Xml.XmlNodeReader.Tests/XmlNodeReaderConstructionTests.cs
Comment (stephentoub) [inline src/libraries/System.Private.Xml/tests/XmlNodeReader/System.Xml.XmlNodeReader.Tests/XmlNodeReaderConstructionTests.cs:26]:
```C#
Assert.Throws<ArgumentNullException>(() => new XmlNodeReader(null));
```

File: src/libraries/System.Private.Xml/tests/XmlDocument/XmlDocumentTests/NameTableTests.cs
Comment (stephentoub) [inline src/libraries/System.Private.Xml/tests/XmlDocument/XmlDocumentTests/NameTableTests.cs:34]:
```suggestion
            new XmlDocument(customNameTable);
```

File: src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializationReader.cs
Comment (MattKotsenas) [reply src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializationReader.cs]: Fixed!

File: src/libraries/System.Private.Xml/src/System/Xml/Xsl/Xslt/CompilerScopeManager.cs
Comment (MattKotsenas) [reply src/libraries/System.Private.Xml/src/System/Xml/Xsl/Xslt/CompilerScopeManager.cs]: Fixed!

File: src/libraries/System.Private.Xml/src/System/Xml/Xsl/Xslt/XsltLoader.cs
Comment (MattKotsenas) [reply src/libraries/System.Private.Xml/src/System/Xml/Xsl/Xslt/XsltLoader.cs]: Fixed!

File: src/libraries/System.Private.Xml/src/System/Xml/Xsl/Xslt/XsltLoader.cs
Comment (MattKotsenas) [reply src/libraries/System.Private.Xml/src/System/Xml/Xsl/Xslt/XsltLoader.cs]: Fixed!

File: src/libraries/System.Private.Xml/tests/XmlDocument/XmlDocumentTests/CreateCDataSectionTests.cs
Comment (MattKotsenas) [reply src/libraries/System.Private.Xml/tests/XmlDocument/XmlDocumentTests/CreateCDataSectionTests.cs]: Fixed!

File: src/libraries/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs
Comment (MattKotsenas) [reply src/libraries/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs]: Fixed!

File: src/libraries/System.Private.Xml/tests/XmlNodeReader/System.Xml.XmlNodeReader.Tests/XmlNodeReaderConstructionTests.cs
Comment (MattKotsenas) [reply src/libraries/System.Private.Xml/tests/XmlNodeReader/System.Xml.XmlNodeReader.Tests/XmlNodeReaderConstructionTests.cs]: Fixed!

File: src/libraries/System.Private.Xml/tests/XmlDocument/XmlDocumentTests/NameTableTests.cs
Comment (MattKotsenas) [reply src/libraries/System.Private.Xml/tests/XmlDocument/XmlDocumentTests/NameTableTests.cs]: Fixed!

File: src/libraries/System.Private.Xml/src/System/Xml/Schema/ContentValidator.cs
Comment (krwq) [inline src/libraries/System.Private.Xml/src/System/Xml/Schema/ContentValidator.cs:1541]: I've occasionally found the DEBUG code useful when debugging some cryptic bugs in XML. I'd prefer we left it in place

File: src/libraries/System.Private.Xml/src/System/Xml/Schema/SchemaCollectionCompiler.cs
Comment (krwq) [inline src/libraries/System.Private.Xml/src/System/Xml/Schema/SchemaCollectionCompiler.cs:2442]: I'd prefer we left this debug code in here as it is occasionally useful

File: src/libraries/System.Private.Xml/src/System/Xml/Schema/ContentValidator.cs
Comment (MattKotsenas) [reply src/libraries/System.Private.Xml/src/System/Xml/Schema/ContentValidator.cs]: Restored. Thanks for the feedback!

File: src/libraries/System.Private.Xml/src/System/Xml/Schema/SchemaCollectionCompiler.cs
Comment (MattKotsenas) [reply src/libraries/System.Private.Xml/src/System/Xml/Schema/SchemaCollectionCompiler.cs]: Restored! Thanks for the feedback!

**Comment by MattKotsenas** [comment]
I don't have merge permissions, so if everything looks good, please merge on my behalf. If there are any issues or questions, please let me know. Thanks!

**Review by krwq — APPROVED** [review]

**Comment by krwq** [comment]
Thanks @MattKotsenas, looks good to me, let's see if we can get clean CI run first

**Comment by danmosemsft** [comment]
@krwq  you can set auto-merge label in such a case of course

**Comment by krwq** [comment]
Thanks @MattKotsenas!

**Pull request merged**
